testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java

```diff
@@ -259,23 +262,25 @@ public void testOfflineSessionLoadingAfterCacheRemoval() {
         reinitializeKeycloakSessionFactory();
         assertOfflineSessionsExist(realmId, offlineSessionIds);
 
-        // remove sessions from the cache
-        withRealm(realmId, (session, realm) -> {
-            // Delete local user cache (persisted sessions are still kept)
-            UserSessionProvider provider = session.getProvider(UserSessionProvider.class);
-            // Remove in-memory representation of the offline sessions
-            if (provider instanceof InfinispanUserSessionProvider) {
-                ((InfinispanUserSessionProvider) provider).removeLocalUserSessions(realm.getId(), true);
-            } else if (provider instanceof PersistentUserSessionProvider) {
-                ((PersistentUserSessionProvider) provider).removeLocalUserSessions(realm.getId(), true);
-            } else if (provider instanceof RemoteUserSessionProvider) {
-                //no-op, session not local
-            } else {
-                throw new IllegalStateException("Unknown UserSessionProvider: " + provider);
-            }
+        if (!InfinispanUtils.isEmbeddedInfinispan()) {
```

Comment:
The condition is wrong, it should be:
```suggestion
        if (InfinispanUtils.isEmbeddedInfinispan()) {
```

The test is checking whether sessions exist after restart and they are lazily loaded if not present in cache. When multi-site or remote Infinispan is enabled this can be skipped because embedded caches are not used.

Reply:
In that case, the branch `provider instanceof RemoteUserSessionProvider` can be removed since it will no longer be `true`.

Reply:
Ok done. I added an assumption to not run when embedded ISPN is not used as the test does not make sense for other profiles.